### PR TITLE
connectors[slack]: update content-nodes to use new itnernalIds

### PR DESCRIPTION
**Pull request by spolu**
## Description

Move connectors to use the new internalIds

Updates:
- retrievePermissions
- setPermissions
- retrieveBatchContentNodes
- retrieveContentNodeParents

## Risk

Break knowledge and assistant builder for slack

## Deploy Plan

- deploy `connectors` 
- shortly after run migrations that changes agent configurations

### CODE REVIEW

File: connectors/src/connectors/slack/index.ts [hunk 1]
```diff
@@ -580,13 +590,13 @@ export class SlackConnectorManager extends BaseConnectorManager<SlackConfigurati
     const channels = await SlackChannel.findAll({
       where: {
         connectorId: this.connectorId,
-        slackChannelId: internalIds,
+        slackChannelId: slackChannelIds,
```
Comment (aubin-tchoi) [inline connectors/src/connectors/slack/index.ts:593]: isn't that missing an Op.in?

File: connectors/src/connectors/slack/index.ts [hunk 2]
```diff
@@ -440,7 +444,9 @@ export class SlackConnectorManager extends BaseConnectorManager<SlackConfigurati
       await SlackChannel.findAll({
         where: {
           connectorId: this.connectorId,
-          slackChannelId: Object.keys(permissions),
+          slackChannelId: Object.keys(permissions).map((k) =>
+            slackChannelIdFromInternalId(k)
+          ),
```
Comment (aubin-tchoi) [inline connectors/src/connectors/slack/index.ts:449]: isn't that missing an Op.in?

File: connectors/src/connectors/slack/index.ts [hunk 3]
```diff
@@ -580,13 +590,13 @@ export class SlackConnectorManager extends BaseConnectorManager<SlackConfigurati
...
         connectorId: this.connectorId,
-        slackChannelId: internalIds,
+        slackChannelId: slackChannelIds,
       },
     });
 
     const contentNodes: ContentNode[] = channels.map((ch) => ({
       provider: "slack",
-      internalId: ch.slackChannelId,
+      internalId: internalIdFromSlackChannelId(ch.slackChannelId),
```
Comment (aubin-tchoi) [inline connectors/src/connectors/slack/index.ts:599]: is it the right time to add a Resource with a method `toContentNode`?

Comment (aubin-tchoi) [reply connectors/src/connectors/slack/index.ts:599]: actually if this moves to core maybe not

**Review by aubin-tchoi — APPROVED** [review]

**Comment by aubin-tchoi** [comment]
Should we also stop sending both the old and new parents in the same PRs?

File: connectors/src/connectors/slack/index.ts (hunk 2 above)
Comment (spolu) [reply connectors/src/connectors/slack/index.ts:449]: Not needed. Also if it worked, it'll work :-)

File: connectors/src/connectors/slack/index.ts (hunk 3 above)
Comment (spolu) [reply connectors/src/connectors/slack/index.ts:599]: Yep this is all going away :-)

**Comment by spolu** [comment]
As per irl will do separately

**Pull request merged**